src/Internal/Payment/Response/ResponseInterface.php

```diff
@@ -0,0 +1,46 @@
+<?php
+/**
+ * Interface ResponseInterface
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay\Internal\Payment\Response;
+
+/**
+ * Payment processing response interface, used to
+ * describe the response of processing a payment.
+ */
+interface ResponseInterface {
```

Comment:
> Why would we want the gateway to be unaware of the states? 

I think that ideally the gateway should care only about the response from the service, rather than its internals and states. For me, this is a way to decouple them.

> With responses, we make the states aware of the gateway.

I cannot agree with this statement. They are an independent set of classes, representing the outcome of the payment process, and are unaware both of the gateway, and of the internal workings of the payment process/context/state.